dev/model.py

```diff
@@ -0,0 +1,22 @@
+from django.db import models
+
+class Location(models.model):
+    location = models.ForeignKey('self', on_delete=models.SET_NULL)
+    name = models.CharField(max_length=30)
+    note = models.TextField()
+
+class Asset(models.model):
+    name = models.CharField(max_length=30)
+    note = models.TextField()
+    metadata = models.JSONField()
```

Comment:
For metadata I wouldn't use a JSON field. They are quite useful for some stuff, but two Tables with relation between them seems more flexible:
```
class MetadataKey(models.model):
    name = models.CharField()

class MetadataItem(models.model):
    assets = models.ForeignKey(Asset)
    key = models.ForeignKey(MetadataKey)
    value = models.CharField(max_length=512)
```

On this way it is easier to generate lists of metadata keys and this is useful for filtering items or generate suggestions.

Reply:
I would also prefer the way mentioned by @clerie, as it also allows easier filtering